src/Microsoft.TemplateEngine.Abstractions/IValidationEntry.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.TemplateEngine.Abstractions
+{
+    /// <summary>
+    /// Represents a validation error when loading the template.
+    /// </summary>
+    public interface IValidationEntry
+    {
+        /// <summary>
+        /// Error severity.
+        /// </summary>
+        public enum SeverityLevel
+        {
+            None,
+            Error,
+            Warning,
+            Info
+        }
+
+        /// <summary>
+        /// Gets the error severity.
+        /// </summary>
+        SeverityLevel Severity { get; }
+
+        /// <summary>
+        /// Gets the code of the error.
+        /// </summary>
+        string Code { get; }
+
+        /// <summary>
+        /// Gets the error message (localized when available).
+        /// </summary>
+        string ErrorMessage { get; }
+
+        /// <summary>
+        /// Gets the details about error location.
+        /// </summary>
+        ErrorLocation? Location { get; }
+
+        /// <summary>
+        /// Details of the error location.
+        /// </summary>
+        public readonly struct ErrorLocation
```

Comment:
Like this!

How does it work when the position in file is not known/spwcifiable? Are the integer members set to zero?

Reply:
Yes, the plan to have them on defaults (0,0). The initial position in the file is usually (1,1), so it is different.